mrmustard/physics/fock.py

```diff
@@ -247,7 +247,13 @@ def fidelity(state_a, state_b, a_ket: bool, b_ket: bool) -> Scalar:
             math.sum(math.conj(b) * math.matvec(math.reshape(state_a, (len(b), len(b))), b))
         )
 
-    raise NotImplementedError("Fidelity between mixed states is not implemented yet.")
+    # mixed state
+    return (
+        math.trace(
+            math.sqrtm(math.matmul(math.matmul(math.sqrtm(state_a), state_b), math.sqrtm(state_a)))
+        )
+        ** 2
+    )
```

Comment:
Looks good, but it seems like @elib20 was "hoping for something with a faster implementation". Maybe this is exactly what he meant, but I think it's worth checking. 🙂 

Still nice to have support for mixed state calculations though.

Reply:
Yes, this is just the same approach @elib20 was following.

I've been thinking on ways to improve performance:one could be implementing the calculation with numba. However, because the calculation is executed by the backend (TF or torch in this case) I don't see much gain on going to numba.

What do you think? do you have any guess on how that can be improved.